### DISCUSSION TITLE
How does web rendering work?

### DISCUSSION THREAD

**Discussion by ghost** [hecrj/iced#1224] [Q&A]
How does Iced target the web? I'm assuming the `glow` renderer is responsible for that, since `wgpu` takes care of native rendering, right?

**Comment by 13r0ck** [comment]
Technically yes, but it is VERY experimental, ~~and considered out of scope~~ (see below).

Iced web is neither glow nor wgpu, iced_web was its own thing that used dodrio to render to the Dom. Since dodrio was archived, that put iced web at a stand still until that is resolved, but writing a web renderer for iced is not a priority for iced, nor will it be for a very long time.

If you want to write a web front end with the elm architecture I recommend learning Elm, it actually is a delightful language.

**Reply by hecrj** [reply]
It's not completely out of scope! Simply not supported at the moment.

**Reply by 13r0ck** [reply]
Thanks for the correction!

**Reply by jg2562** [reply]
What caused the drop of support? Was it the depreciation of dodrio?

**Reply by ghost** [reply]
@hecrj what's the plan here? Are you interested in bringing the web as a compilation target to Iced? I believe this would be a formidable goal (in a good way). Cross-platform GUI applications that utilize native renders when installed locally and Wasm/WebGPU when deployed to the web. Same UI everywhere + high performance. One source code, written in Rust.

**Reply by hecrj** [reply]
@jg2562 Yes, although we haven't dropped support. We just moved the crate to a different repository: https://github.com/iced-rs/iced_web. The idea is to build `iced_web` (DOM support) on top of `iced` itself instead of the other way around.

@leontepe That is exactly what landed in #1096.

**Reply by jg2562** [reply]
@hecrj Wouldn't that mean web is a secondary target? Thus support of it has to be opted into?

**Reply by hecrj** [reply]
@jg2562 If anything, the latest changes in #1096 make targetting the Web platform easier! See my final comment here: https://github.com/iced-rs/iced/pull/1096#pullrequestreview-866907637

In any case, web has always been a secondary target. `iced_web` is very experimental and only supports a subset of the native widgets.

**Reply by jg2562** [reply]
Oh I understand now, I like how it's separated then. It also means I need to move an issue off of `iced_web` and into `iced`. Thank you for clearing that up. I was planning on looking into making events work for `iced_web` but I think ill try to make it work for `iced` when compiled for `wasm` then. Thanks!
